wordpress-mobile/WordPress-Android
# [Edit Comment] Remove Legacy `EditCommentActivity` Screen

Related To https://github.com/wordpress-mobile/WordPress-Android/issues/18906#issuecomment-1725530676 & https://github.com/wordpress-mobile/WordPress-Android/issues/18906#issuecomment-1731442062

This PR deletes the legacy and used [EditCommentActivity.java](https://github.com/wordpress-mobile/WordPress-Android/blob/8ce321d5e9f6eaed0adf8b2128da720d5e365292/WordPress/src/main/java/org/wordpress/android/ui/comments/EditCommentActivity.java#L4) class, by first deleting the no longer necessary [UnifiedCommentsCommentEditFeatureConfig.kt](https://github.com/wordpress-mobile/WordPress-Android/blob/8ce321d5e9f6eaed0adf8b2128da720d5e365292/WordPress/src/main/java/org/wordpress/android/util/config/UnifiedCommentsCommentEditFeatureConfig.kt) feature flag.

-----

## To test:

1. Quickly smoke test any edit comment related functionality on both, the WordPress and Jetpack apps, and see if everything is working as expected.

<details>
    <summary>1. Edit Comment Screen from Comments Screen [UnifiedCommentsEditActivity.java]</summary>

ℹ️ This test applies to both, the `Jetpack` and `WordPress` app.

- Go to `Comments` screen and tap on any comment.
- From the `More` on the right side, at the bottom of the comment, select `Edit`.
- Verify that the `Edit Comment` screen is shown and functioning as expected.
- For example, try editing/adding the comment's `Name`, `Web address`, `Email address` and its
`Comment` itself.

</details>

<details>
    <summary>2. Edit Comment Screen from Notification Tab [UnifiedCommentsEditActivity.java]</summary>

ℹ️ This test applies to the `Jetpack` app only.

- Go to `Notifications` tab, then its `COMMENTS` sub-tab, and tap on any comment.
- From the `More` on the right side, at the bottom of the comment, select `Edit`.
- Verify that the `Edit Comment` screen is shown and functioning as expected.
- For example, try editing/adding the comment's `Name`, `Web address`, `Email address` and its
`Comment` itself.

</details>

-----

## Regression Notes

1. Potential unintended areas of impact

    - N/A

3. What I did to test those areas of impact (or what existing automated tests I relied on)

    - See `To test` section above.

4. What automated tests I added (or what prevented me from doing so)

    - N/A

-----

## PR submission checklist:


-----

## UI Changes testing checklist:
